### PR TITLE
Made pypo.pounit encoding agnostic

### PR DESCRIPTION
**Pull request by claudep**
This is a follow-up commit of 0199313bd63.

### CODE REVIEW

File: translate/storage/pypo.py [hunk 1]
```diff
@@ -35,7 +35,6 @@
 from translate.misc.deprecation import deprecated
 from translate.misc.multistring import multistring
 from translate.storage import base, pocommon, poparser
-from translate.storage.pocommon import encodingToUse
```
Comment (unho) [inline translate/storage/pypo.py]: If I am not wrong this is the last usage of `encodingToUse` so perhaps it can also removed from codebase.

**Comment by claudep** [comment]
You're right. I removed that function and rebased the patch.

File: translate/storage/test_pypo.py [hunk 2]
```diff
@@ -261,17 +262,17 @@ def test_output_str_unicode(self):
         assert len(pofile.units) == 1
         print(pofile.serialize())
         thepo = pofile.units[0]
-        assert str(thepo) == posource.encode("UTF-8")
-        # extra test: what if we set the msgid to a unicode? this happens in prop2po etc
-        thepo.source = u"Norwegian Bokm\xe5l"
```
Comment (unho) [inline translate/storage/test_pypo.py]: Why is this extra test removed?

Comment (claudep) [inline translate/storage/test_pypo.py]: Because with the new strategy, thepo.source is already a unicode string and setting `thepo.source` to an unicode string is not doing anything worth testing, IMHO.

**Comment by unho** [comment]
Merged. Thanks!
